src/scanner.rs

```diff
@@ -1277,28 +1293,189 @@ pub fn scan_file(
     scan_results: &mut ScanResults,
     sinks: HashSet<IdentPath>,
     enabled_cfg: &HashMap<String, Vec<String>>,
+    expand_macro:bool,
 ) -> Result<()> {
     debug!("Scanning file: {:?}", filepath);
-
+    
     // Load file contents
     let mut file = File::open(filepath)?;
     let mut src = String::new();
     file.read_to_string(&mut src)?;
     let syntax_tree = syn::parse_file(&src)?;
-
     // Initialize resolver
     let file_resolver = FileResolver::new(crate_name, resolver, filepath)?;
-
     // Initialize scanner
     let mut scanner = Scanner::new(filepath, file_resolver, scan_results, enabled_cfg);
     scanner.add_sinks(sinks);
-
-    // Scan file contents
     scanner.scan_file(&syntax_tree);
-
+    if expand_macro {
+        let current_file_id =
+            resolver.find_file_id(&filepath).context("cannot find current file id")?;
+        let syntax = resolver.db().parse_or_expand(current_file_id.into());
+        let file_resolver_expand = FileResolver::new(crate_name, resolver, filepath)?;
+        let mut expanded_files: Vec<(syn::File, String)> = Vec::new();
+        let mut expanded_items: Vec<(syn::Item, String)> = Vec::new();
+        let mut expanded_stmts: Vec<(syn::Stmt, String)> = Vec::new();
+        let mut expanded_exprs: Vec<(syn::Expr, String)> = Vec::new();
+        let mut expanded_blocks: Vec<(syn::Block, String)> = Vec::new();
```

Comment:
nice work! It might be preferable to factor out some of this in a separate file, `scanner.rs` is getting a bit long

Reply:
can you put this in a separate `macro_expand.rs`?